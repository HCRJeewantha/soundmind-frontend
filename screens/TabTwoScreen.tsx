import { StyleSheet, TextInput, Text, View, SafeAreaView, Modal, StatusBar, Pressable, FlatList, Image } from "react-native";
import { FontAwesome } from '@expo/vector-icons';
import React, { useState } from "react";
import axios from 'axios'
import { LinearGradient } from "expo-linear-gradient";
import Toast from 'react-native-toast-message';

const baseUrl = 'http://127.0.0.1:8000';
let selectedSongName:any;
let selectedSongImg:any;
let selectedSongId:any;
export function TabTwoScreen() {

  const [searchParam, setSearchParam] = useState('');
  const [searchData, onSearchRequest] = useState(null);

  const showToast = (title: any, message: any, type: any) => {
    Toast.show({
      type: type,
      text1: title,
      text2: message
    });
  }

  const setSelectedValues = async (name: any, song_img: any, song_id: any) => {
    selectedSongName = name
    selectedSongImg = song_img
    selectedSongId = song_id
  }

  const search = async (param: any) => {
    axios.get(
      baseUrl + '/request-songs/' + param
    ).then((response) => {
      onSearchRequest(response.data.items)
    }).catch((error) => {
      if (error.response) {
        showToast("Error", error.response.data.detail, "error")
      }
    });
  };

  const addToPlayList = async () => {
    const payload = {
      "name": selectedSongName,
      "song_img": selectedSongImg,
      "song_id": selectedSongId
    }
    axios.post(
      baseUrl + '/add-songs-to-playlist',
      payload
    ).then((response) => {
      showToast("Success", "Song added to playlist", "success")
      setModalVisible(!modalVisible);
    }).catch((error) => {
      if (error.response) {
        showToast("Error", error.response.data.detail, "error")
      }
    });
  };

  const renderItem = ({ item, index }: any) => (
    <View style={{ flex: 1, flexDirection: 'row', borderRadius: 2, width: '100%', backgroundColor: '#b400ff33' }}>
      <View style={{ width: '10%', height: '80px', padding: 5 }} >
        <Text style={styles.number}>{index + 1}</Text>
      </View>
      <View style={{ width: '30%', height: '80px', margin: 5 }} >
        <Image
          style={{
            width: 'auto',
            height: '100%',
            resizeMode: 'contain',
            borderRadius: 10
          }}
          source={item.snippet.thumbnails.default.url} />
      </View>
      <View style={{ width: '45%', height: '80px', padding: 5 }} >
        <Text numberOfLines={2} style={styles.title}>{item.snippet.title}</Text>
      </View>
      <View style={{ width: '10%', height: '80px', display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
        <Pressable
          testID="selectSongBtn"
          style={[styles.button]}
          onPress={() => setSelectedValues(item.snippet.title, item.snippet.thumbnails.high.url, item.id.videoId).then(() => { setModalVisible(true) })}
        >
          <FontAwesome
            name="ellipsis-h"
            size={25}
            style={{ margin: '5px', color: '#fff' }}
          />
        </Pressable>
      </View>
    </View>
  );

  const ModelView = (props: any) => (
    <View style={{ height: 'auto', width: '90%' }}>
      <Modal
        animationType="slide"
        transparent={true}
        visible={modalVisible}
        onRequestClose={() => {
          setModalVisible(!modalVisible);
        }}
      >
        <View style={{ backgroundColor: 'rgba(0, 0, 0, 0.5)', height: '100%' }}>
          <View style={{ position: 'absolute', bottom: '2%', margin: '0%', left: '2%', right: '2%' }}>
            <View style={styles.centeredView}>
              <View style={styles.modalView}>
                <View style={{ width: '100%', height: '180px', marginHorizontal: 5, marginVertical: 40 }}>
                  <Image
                    style={{
                      width: '100%',
                      height: '100%',
                      resizeMode: 'contain',
                      borderRadius: 10
                    }}
                    source={props.img} />
                </View>
                <View style={{ width: "96%" }}>
                  <Text style={{ fontSize: 20, fontWeight: 'bold', textAlign: 'center' }}>{selectedSongName}</Text>
                </View>
                <View style={{ marginBottom: 20 }}>
                  <Text style={{ fontSize: 17 }}>Playlist | ...</Text>
                </View>
                <View style={{ marginBottom: 20 }}>
                  <Pressable
                    style={{
                      height: 80, width: 80, borderRadius: 100, backgroundColor: '#2196f3', shadowOpacity: 0.25,
                      shadowRadius: 30,
                      shadowOffset: {
                        height: 0,
                        width: 0,
                      },
                      alignItems: 'center',
                      justifyContent: 'center',
                    }}
                    onPress={() => addToPlayList()}
                  >
                    <FontAwesome
                      style={{ marginLeft: 10 }}
                      name="play"
                      size={38}
                      color={'#fff'}
                    />
                  </Pressable>
                </View>
              </View>
              <View style={styles.modelActionButtons}>
                <Pressable
                  style={{ padding: 15 }}
                  onPress={() => addToPlayList()}
                >
                  <Text style={{ color: '#0001ff', fontSize: 18 }}>Add To Playlist</Text>
                </Pressable>
              </View>
              <View style={styles.modelActionButtons}>
                <Pressable
                  style={{ padding: 15 }}
                  onPress={() => setModalVisible(!modalVisible)}
                >
                  <Text style={{ color: '#000', fontSize: 18 }}>Close</Text>
                </Pressable>
              </View>
            </View>
          </View>
        </View>
      </Modal>
    </View>
  )
  const [modalVisible, setVisibleState] = useState(false);
  const setModalVisible = (visible: any) => {
    setVisibleState(visible)
  }

  return (
    <View style={styles.mainContainer}>
      <LinearGradient
        colors={['#000', '#75dadf']}
        style={styles.background}
      />
      <SafeAreaView style={styles.container}>
        <View style={{ flexDirection: 'row', width: '100%' }}>
          <View style={{ width: '80%', height: 'auto' }}>
            <TextInput
              style={styles.input}
              placeholder="Search Songs"
              keyboardType="numeric"
              onChange={(event: any) => setSearchParam(event.target.value)}
            />
          </View>
          <View style={{ width: '20%', height: 'auto', marginLeft: 5 }}>
            <Pressable
              testID="searchBtn"
              style={{
                backgroundColor: '#2196f3', height: 40,
                margin: 12,
                width: "70%",
                borderRadius: 5
              }}
              onPress={() => search(searchParam)}
            >
              <FontAwesome
                name="search"
                size={25}
                style={{ margin: 'auto', color: '#fff' }}
              />
            </Pressable>
          </View>
        </View>
        <FlatList
          data={searchData}
          renderItem={renderItem}
        // keyExtractor={item => item}
        />
        <View style={styles.modelContainer}>
          <ModelView img={selectedSongImg} name={selectedSongName} />
        </View>
      </SafeAreaView>
    </View>
  );
}

const styles = StyleSheet.create({
  input: {
    height: 40,
    margin: 12,
    borderWidth: 1,
    width: "100%",
    padding: 10,
    borderRadius: 5,
    backgroundColor: '#fff'
  },
  modelContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  container: {
    flex: 1,
    paddingTop: StatusBar.currentHeight,
    display: 'flex',
    alignItems: 'center',
  },
  mainContainer: {
    flex: 1,
  },
  background: {
    zIndex: -10,
    position: 'absolute',
    left: 0,
    right: 0,
    top: '50%',
    height: '100%',
  },
  item: {
    backgroundColor: "#f9c2ff",
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
    alignItems: 'flex-start'
  },
  header: {
    fontSize: 32,
    backgroundColor: "#fff"
  },
  title: {
    fontSize: 15,
    color: '#fff',
    fontWeight: '600',
    marginTop: '5px'
  },
  number: {
    fontSize: 15,
    color: '#fff',
    fontWeight: '600',
    top: '40%',
    position: 'absolute',
    marginLeft: 10
  },
  subtitle: {
    fontSize: 13,
    color: '#fff',
    fontWeight: '400',

  },
  centeredView: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    width: '100%',
    height: 'auto',
    bottom: '0%'
  },
  modalView: {
    backgroundColor: "white",
    borderRadius: 20,
    height: 'auto',
    width: '96%',
    alignItems: "center",
    elevation: 5
  },
  button: {
    borderRadius: 20,
    padding: 10,
    elevation: 2
  },
  buttonClose: {
    backgroundColor: "#2196F3",
  },
  textStyle: {
    color: "white",
    fontWeight: "bold",
    textAlign: "center"
  },
  modalText: {
    marginBottom: 15,
    textAlign: "center"
  },
  modelActionButtons: {
    marginTop: 10, backgroundColor: "white",
    borderRadius: 20,
    height: 'auto',
    width: '96%',
    alignItems: "center",
  }
});
