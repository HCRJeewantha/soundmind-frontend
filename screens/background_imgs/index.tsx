export const SUNSET = require('./sunset.jpg'); 
export const LOGO = require('./logo.png'); 