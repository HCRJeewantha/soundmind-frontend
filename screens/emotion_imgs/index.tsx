export const NEUTRAL = require('./neutral.png'); 
export const HAPPY = require('./happy.png'); 
export const SAD = require('./sad.png'); 
export const FEAR = require('./fear.png'); 
export const ANGRY = require('./angry.png'); 
