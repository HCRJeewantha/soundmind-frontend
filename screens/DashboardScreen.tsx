import React, { useState, useEffect } from "react";
import { Pressable, StyleSheet, TextInput, Text, Image, ScrollView, Dimensions, FlatList } from "react-native";
import { View } from "../components/Themed";
import { RootStackScreenProps, RootTabScreenProps } from "../types";
import { LinearGradient } from 'expo-linear-gradient';
import { LOGO, PLAYLIST } from "./background_imgs";
import { FontAwesome } from "@expo/vector-icons";
import axios from 'axios'

const baseUrl = 'http://127.0.0.1:8000';
const { width } = Dimensions.get('window');

export default function TabFourScreen({ navigation }: RootTabScreenProps<'TabFour'>) {

    const [musicForYou, setMusicForYou] = useState([]);

    const getMusicForYou = async () => {
        axios.get(
            baseUrl + '/get-music-for-you/rock'
        ).then((response) => {
            console.log(response.data.tracks.track)
            setMusicForYou(response.data.tracks.track)
        }).catch((error) => {
            console.log(error)
        })
    }

    useEffect(() => {
        //get playlist
        getMusicForYou()
    }, []);

    const renderItem = ({ item, index }: any) => (
        <View>
            <View style={styles.view} />
        </View>
    );


    return (
        <React.Fragment>
            <LinearGradient
                colors={['#262626', '#75dadf']}
                style={styles.background}
            />
            <ScrollView style={styles.container}>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none', marginVertical: 10 }}>
                    <View style={{ flex: 1, flexDirection: 'row', width: '95%', height: '80px', backgroundColor: 'none' }}>
                        <View style={{ width: '90%', height: '50px', padding: 5, justifyContent: 'center', alignItems: 'flex-start', backgroundColor: 'none' }} >
                            <Text style={{ fontSize: 20, fontWeight: '600', color: '#fff' }}>Good Evening</Text>
                        </View>
                        <View style={{ width: '10%', height: '50px', padding: 5, justifyContent: 'center', alignItems: 'center', backgroundColor: 'none' }} >
                            <FontAwesome
                                style={{ marginRight: 5 }}
                                name="gear"
                                size={20}
                                color={'#fff'}
                            />
                        </View>
                    </View>
                </View>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none', marginVertical: 10 }}>
                    <View style={{ flex: 1, flexDirection: 'row', backgroundColor: '#00000052', width: '95%', borderRadius: 10 }}>
                        <View style={{ width: '30%', height: 'auto', justifyContent: 'center', alignItems: 'flex-start', backgroundColor: 'none' }} >
                            <View style={{ height: 80, width: 90, alignContent: 'center', alignItems: 'center', backgroundColor: 'none' }}>
                                <Image
                                    style={{
                                        borderTopLeftRadius: 10,
                                        borderBottomLeftRadius: 10,
                                        width: '100%',
                                        height: '100%',
                                        resizeMode: 'stretch'
                                    }}
                                    source={PLAYLIST} />
                            </View>
                        </View>
                        <View style={{ width: '90%', height: 'auto', padding: 5, justifyContent: 'center', alignItems: 'flex-start', backgroundColor: 'none' }} >
                            <Text style={{ fontSize: 20, fontWeight: '600', color: '#fff' }}>My Playlist</Text>
                        </View>
                    </View>
                </View>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none', marginVertical: 10 }}>
                    <View style={{ flex: 1, flexDirection: 'row', width: '95%', backgroundColor: 'none' }}>
                        <Text style={{ fontSize: 20, fontWeight: '600', color: '#fff' }}>Music For You</Text>
                    </View>
                </View>
                <FlatList
                    ref={(scrollView) => { scrollView = scrollView; }}
                    style={styles.container}
                    horizontal={true}
                    decelerationRate={0}
                    snapToInterval={width - 60}
                    snapToAlignment={"center"}
                    data={musicForYou}
                    renderItem={renderItem}
                    contentInset={{
                        top: 0,
                        left: 30,
                        bottom: 0,
                        right: 30,
                    }} />
            </ScrollView>
        </React.Fragment>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    linearGradient: {
        flex: 1,
        paddingLeft: 15,
        paddingRight: 15,
        borderRadius: 5
    },
    background: {
        position: 'absolute',
        left: 0,
        right: 0,
        top: 0,
        height: '100%',
    },
    view: {
        backgroundColor: 'blue',
        width: width - 80,
        margin: 10,
        height: 200,
        borderRadius: 10,
    },
    view2: {
        backgroundColor: 'red',
        width: width - 80,
        margin: 10,
        height: 200,
        borderRadius: 10,
    },
});

