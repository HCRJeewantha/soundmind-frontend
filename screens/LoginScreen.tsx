import React, { useState } from "react";
import { Pressable, StyleSheet, TextInput, Text, Image } from "react-native";
import { View } from "../components/Themed";
import { LinearGradient } from 'expo-linear-gradient';
import { LOGO } from "./background_imgs";
import axios from 'axios'
import Toast from 'react-native-toast-message';

//Base URL
const baseUrl = 'http://127.0.0.1:8000';

export function LoginScreen({ navigation }: any) {

    const [mobile, setMobile] = useState('');
    const [password, setPassword] = useState('');

    const showToast = (title: any, message: any, type: any) => {
        Toast.show({
            type: type,
            text1: title,
            text2: message
        });
    }

    const validations = () => {
        let formIsValid = true;
        if (!mobile) {
            showToast("Error", "Mobile Number is required", "error")
            formIsValid = false;
        }
        if (mobile.length > 11) {
            showToast("Error", "Mobile Number shoud less than 10", "error")
            formIsValid = false;
        }
        if (!password) {
            showToast("Error", "Password required", "error")
            formIsValid = false;
        }
        return formIsValid;
    }

    const signIn = async () => {
        if (validations()) {
            const payload = {
                "mobile": mobile,
                "password": password
            }

            axios.post(
                baseUrl + '/sign-in',
                payload
            ).then((response) => {
                localStorage.setItem("id", response.data.id)
                localStorage.setItem("name", response.data.name)
                showToast("Success", "Login Successful", "success")
                navigation.navigate('Welcome')
            }).catch((error) => {
                if (error.response) {
                    showToast("Error", error.response.data.detail, "error")
                }
            });
        }
    };

    return (
        <View style={styles.container}>
            <LinearGradient
                colors={['#262626', '#75dadf']}
                style={styles.background}
            />
            <View style={{ width: '100%', height: 'auto', bottom: 0, position: 'absolute', backgroundColor: '#ffffff69', borderTopLeftRadius: 20, borderTopRightRadius: 20 }}>

                <View style={{ height: 120, width: 120, borderRadius: 100, alignContent: 'center', alignItems: 'center', margin: 'auto', zIndex: 1, top: -50, backgroundColor: '#ffffff69' }}>
                    <Image
                        style={{
                            width: '100%',
                            height: '100%',
                            resizeMode: 'stretch'
                        }}
                        source={LOGO} />
                </View>
                <Text style={{ marginLeft: '10%', color: '#4a4141', fontWeight: '700', fontSize: 22, marginTop: 40 }}>Signin</Text>
                <Text style={{ marginLeft: '10%', color: '#4a4141', fontWeight: '600', fontSize: 15, marginTop: 20 }}>Mobile Number</Text>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none' }}>
                    <View
                        style={{
                            width: '80%',
                            height: '50px',
                            borderRadius: 10,
                            marginTop: '10px',
                            shadowOpacity: 0.25,
                            shadowRadius: 30,
                            shadowOffset: {
                                height: 0,
                                width: 0,
                            },
                        }}>
                        <TextInput
                            onChange={(event: any) => setMobile(event.target.value)}
                            secureTextEntry={false}
                            style={{
                                height: '100%',
                                padding: '10px',
                            }}
                            placeholder="07xxxxxx"
                        />
                    </View>
                </View>
                <Text style={{ marginLeft: '10%', color: '#4a4141', fontWeight: '600', fontSize: 15, marginTop: 20 }}>Password</Text>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none' }}>
                    <View
                        style={{
                            width: '80%',
                            height: '50px',
                            borderRadius: 10,
                            marginTop: '10px',
                            shadowOpacity: 0.25,
                            shadowRadius: 30,
                            shadowOffset: {
                                height: 0,
                                width: 0,
                            },
                        }}>
                        <TextInput
                            onChange={(event: any) => setPassword(event.target.value)}
                            secureTextEntry={true}
                            style={{
                                height: '100%',
                                padding: '10px',
                            }}
                            placeholder="password"
                        />
                    </View>
                </View>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none', marginVertical: 15 }}>
                    <Pressable onPress={() => navigation.navigate('Registration')}>
                        <Text style={{ color: '#4a4141', fontWeight: '500', fontSize: 15 }}>Don't you have an account? <b>Sign up</b></Text>
                    </Pressable>
                </View>
                <View style={{ width: '100%', height: 'auto', justifyContent: 'center', alignItems: 'center', backgroundColor: 'none', marginTop: 10, marginBottom: 40 }}>
                    <Pressable
                        testID="signInBtn"
                        style={{
                            alignItems: 'center',
                            justifyContent: 'center',
                            width: '100%',
                        }}
                        onPress={() => signIn()}>
                        <LinearGradient
                            colors={['#50aeffeb', '#26ced7f2']}
                            style={{
                                marginTop: 30,
                                borderRadius: 10,
                                borderWidth: 1,
                                borderColor: '#2196f3',
                                alignItems: 'center',
                                justifyContent: 'center',
                                width: '80%',
                                height: '50px',
                                backgroundColor: '#2196f3',
                                shadowOpacity: 0.25,
                                shadowRadius: 30,
                                shadowOffset: {
                                    height: 0,
                                    width: 0,
                                },
                            }}>
                            <Text style={{ color: '#fff', fontSize: 20, fontWeight: '600' }}>Sign In</Text>
                        </LinearGradient>
                    </Pressable>
                </View>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    linearGradient: {
        flex: 1,
        paddingLeft: 15,
        paddingRight: 15,
        borderRadius: 5
    },
    background: {
        position: 'absolute',
        left: 0,
        right: 0,
        top: 0,
        height: '100%',
    },
});

