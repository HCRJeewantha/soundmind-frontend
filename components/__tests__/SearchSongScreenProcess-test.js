import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { render, fireEvent } from '@testing-library/react-native';
import { TabTwoScreen } from '../../screens/TabTwoScreen';
import renderer from 'react-test-renderer';

describe('Testing search song by name', () => {

    /*
    Snapshot Testing
    */
    test("Search song UI render correctly", async () => {
        const tree = renderer.create(<TabTwoScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    /*
    Intergration Testing
   */
    test('Enter a name into text field and click search button', async () => {
        const component = (
            <NavigationContainer>
                <TabTwoScreen />
            </NavigationContainer>
        );

        const { findByTestId, getByPlaceholderText } = render(component);
        fireEvent.changeText(getByPlaceholderText('Search Songs'), 'Faded');
        const toClick = await findByTestId("searchBtn");
        fireEvent(toClick, 'press');

    });

    test('Clicking on one song takes you to the details screen', async () => {
        const component = (
                <TabTwoScreen />
        );
        const { queryByTestId } = render(component);
        const toClick = await queryByTestId("selectSongBtn");
        // fireEvent(toClick, 'press');

        // const newBody = await findByText('the number you have chosen is 5');
        // expect(newBody).toBeTruthy();
    });
});