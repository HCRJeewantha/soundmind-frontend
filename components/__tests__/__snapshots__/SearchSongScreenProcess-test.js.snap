// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing search song by name Search song UI render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4278190080,
          4285913823,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Object {
        "height": "100%",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": "50%",
        "zIndex": -10,
      }
    }
  />
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flex": 1,
        "paddingTop": null,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "height": "auto",
            "width": "80%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          keyboardType="numeric"
          onChange={[Function]}
          placeholder="Search Songs"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 40,
              "margin": 12,
              "padding": 10,
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        style={
          Object {
            "height": "auto",
            "marginLeft": 5,
            "width": "20%",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#2196f3",
              "borderRadius": 5,
              "height": 40,
              "margin": 12,
              "width": "70%",
            }
          }
          testID="searchBtn"
        >
          <Text />
        </View>
      </View>
    </View>
    <RCTScrollView
      data={null}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "height": "auto",
            "width": "90%",
          }
        }
      >
        <Modal
          animationType="slide"
          hardwareAccelerated={false}
          onRequestClose={[Function]}
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.5)",
                "height": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "bottom": "2%",
                  "left": "2%",
                  "margin": "0%",
                  "position": "absolute",
                  "right": "2%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "bottom": "0%",
                    "flex": 1,
                    "height": "auto",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderRadius": 20,
                      "elevation": 5,
                      "height": "auto",
                      "width": "96%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "height": "180px",
                        "marginHorizontal": 5,
                        "marginVertical": 40,
                        "width": "100%",
                      }
                    }
                  >
                    <Image
                      style={
                        Object {
                          "borderRadius": 10,
                          "height": "100%",
                          "resizeMode": "contain",
                          "width": "100%",
                        }
                      }
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "width": "96%",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "fontSize": 20,
                          "fontWeight": "bold",
                          "textAlign": "center",
                        }
                      }
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "marginBottom": 20,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "fontSize": 17,
                        }
                      }
                    >
                      Playlist | ...
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "marginBottom": 20,
                      }
                    }
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#2196f3",
                          "borderRadius": 100,
                          "height": 80,
                          "justifyContent": "center",
                          "shadowOffset": Object {
                            "height": 0,
                            "width": 0,
                          },
                          "shadowOpacity": 0.25,
                          "shadowRadius": 30,
                          "width": 80,
                        }
                      }
                    >
                      <Text />
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderRadius": 20,
                      "height": "auto",
                      "marginTop": 10,
                      "width": "96%",
                    }
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "padding": 15,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#0001ff",
                          "fontSize": 18,
                        }
                      }
                    >
                      Add To Playlist
                    </Text>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderRadius": 20,
                      "height": "auto",
                      "marginTop": 10,
                      "width": "96%",
                    }
                  }
                >
                  <View
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "padding": 15,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#000",
                          "fontSize": 18,
                        }
                      }
                    >
                      Close
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </Modal>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
